### PR TITLE
Python: fix nice locations for import aliases

### PR DESCRIPTION
**Pull request by yoff**
The computation of nice location for import statements did not take aliases into account.
This has causes problems for some of our downstream CI.

See also https://github.com/github/codeql-team/issues/2260

### CODE REVIEW

**Comment by RasmusWL** [comment]
I had to look up what `NiceLocationExpr` even was, since I had never heard about it before. It seems to only be used in our jump-to-definition queries... although the _reason_ for needing these is a bit obscure to me, it seems to be to ensure spans for things we have definitions for are not overlapping :thinking: 

For reference, here is how it works for ImportMember:

### Using Expr

image

ImportMember covers whole `resource_provider as rp_obj`

### Using NiceLocaitonExpr

image

ImportMember only covers `rp_obj`

File: python/ql/lib/change-notes/2023-08-10-fix-alias-locations.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: fix
+---
+* Fixed the computation of display locations for imports with aliases.
```
Comment (RasmusWL) [inline python/ql/lib/change-notes/2023-08-10-fix-alias-locations.md:4]:
```suggestion
* Fixed the computation of locations for imports with aliases in jump-to-definition.
```

**Review by RasmusWL — APPROVED** [review]
Minor suggestion for change note :blush: 

Did you test out jump-to-def locally before/after changes? (just curious, I think this change is fine to merge evne without doing this)

**Comment by yoff** [comment]
> For reference, here is how it works for ImportMember:

Those screenshots are with the new computation, right?

**Comment by yoff** [comment]
> Did you test out jump-to-def locally before/after changes?

We (a large part of this was done together with `@esbena`) fiddled a bit with it, but could not see a difference (should it have a different hit-box or what?).

**Comment by RasmusWL** [comment]
> Those screenshots are with the new computation, right?

yes :+1:

> > Did you test out jump-to-def locally before/after changes?
> 
> We (a large part of this was done together with `@esbena`) fiddled a bit with it, but could not see a difference (should it have a different hit-box or what?).

I see, thanks :+1:

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**